reference/ReferenceSeaport.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import { Consideration } from "./lib/ReferenceConsideration.sol";
+
+contract ReferenceSeaport is ReferenceConsideration {
+    /**
+     * @notice Derive and set hashes, reference chainId, and associated domain
+     *         separator during deployment.
+     *
+     * @param conduitController A contract that deploys conduits, or proxies
+     *                          that may optionally be used to transfer approved
+     *                          ERC20/721/1155 tokens.
+     */
+    constructor(address conduitController) Consideration(conduitController) {}
+
+    /**
+     * @dev Internal pure function to retrieve and return the name of this
+     *      contract.
+     *
+     * @return The name of this contract.
+     */
+    function _name() internal pure override returns (string memory) {
+        // Return the name of the contract.
+        assembly {
```

Comment:
Since this is the reference implementation we can just return `_nameString()` here and not use assembly!